web/profiles/custom/yalesites_profile/modules/custom/ys_core/ys_core.tokens.inc

```diff
@@ -51,6 +55,14 @@ function ys_core_tokens($type, array $tokens, array $data, array $options, Bubbl
           $config = \Drupal::config('ys_core.site');
           $replacements[$original] = $config->get('page.events') ?: '/events';
           break;
+
+        case 'teaser_title_node_title':
+          if (!empty($data['node'])) {
+            $nodeTitle = $data['node']->getTitle();
+            $teaserTitle = ($data['node']->field_teaser_title->first()) ? $data['node']->field_teaser_title->first()->getString() : NULL;
```

Comment:
This works fine but may not be future proof. `($data['node']->field_teaser_title->first())` will throw a fatal error if a node does not have a the teaser title field -- as you can not call the first method on null. But I *think* if you change this expression to `($data['node']->field_teaser_title)` it will fail gracefully.

Reply:
Very good catch. Turns out we need to do both - check for the field to exist, but also to check for the `first()` as if that doesn't exist, no teaser text is inputed. Fixed and re-review.